### PR TITLE
ARROW-15653: [R][CI] Fix tests of bundled cpp source

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

**Comment by nealrichardson** [comment]
@github-actions crossbow submit test-r-linux-as-cran test-r-offline*

File: ci/scripts/r_docker_configure.sh [hunk 1]
```diff
@@ -45,27 +56,22 @@ fi
...
-  elif [ "`which zypper`" ]; then
-    zypper install -y libcurl-devel libopenssl-devel
-  else
+  if [ "$PACKAGE_MANAGER" = "apt-get" ]; then
     apt-get update
     apt-get install -y libcurl4-openssl-dev libssl-dev
+  else
+    $PACKAGE_MANAGER install -y libcurl-devel openssl-devel
   fi
+fi
```
Comment (jonkeane) [inline ci/scripts/r_docker_configure.sh:71]:
```suggestion
```

I think this is covered by the `fi` on line 84 (nee 78)

File: dev/tasks/r/github.linux.cran.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ jobs:
 
       - name: Docker Run
         shell: bash
-        run: archery docker run r
+        run: archery docker run -e ARROW_SOURCE_HOME='' r
```
Comment (jonkeane) [inline dev/tasks/r/github.linux.cran.yml:47]:
```suggestion
        # setting ARROW_SOURCE_HOME='' here ensures that we use the synced cpp source
        run: archery docker run -e ARROW_SOURCE_HOME='' r
```

**Comment by nealrichardson** [comment]
@github-actions crossbow submit test-r-linux-as-cran test-r-offline*

File: ci/scripts/r_docker_configure.sh [hunk 3]
```diff
@@ -30,6 +30,17 @@ fi
...
 
+# Figure out what package manager we have
+if [ "`which dnf`" ]; then
+  PACKAGE_MANAGER=dnf
+elif [ "`which yum`" ]; then
+  PACKAGE_MANAGER=yum
+elif [ "`which zypper`" ]; then
+  PACKAGE_MANAGER=zypper
+else
+  PACKAGE_MANAGER=apt-get
```
Comment (jonkeane) [inline ci/scripts/r_docker_configure.sh:41]: Should we move the `apt-get update` up here from down in the `if` below? 

I suspect we don't have a test that has `ARROW_S3 == OFF` and `ARROW_R_DEV == FALSE`, but if we did the update wouldn't have happened there (line 66) so line 83 won't have the update yet

**Review by jonkeane — APPROVED** [review]
Looks good, one small change, though I don't think we have a configuration which would run into this issue

File: ci/scripts/r_docker_configure.sh (hunk 3 above)
Comment (nealrichardson) [reply ci/scripts/r_docker_configure.sh:41]: yeah probably so

**Pull request closed without merging**